hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetInfoHandler.java

```diff
@@ -101,8 +107,27 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
     private @Nullable Optional<FileInfo> infoForFile(
             @NonNull final FileID fileID,
             @NonNull final ReadableFileStore fileStore,
-            @NonNull final LedgerConfig ledgerConfig) {
-        final var meta = fileStore.getFileMetadata(fileID);
+            @NonNull final LedgerConfig ledgerConfig,
+            @NonNull final ReadableUpgradeStoreImpl upgradeStore,
+            @NonNull final FilesConfig fileServiceConfig) {
+
+        FileMetadata meta = null;
+        // the update file always will be for the node, not a particular ledger that's why we just compare th num
+        if (fileID.fileNum() == fileServiceConfig.upgradeFileNumber()) {
+            final var file = upgradeStore.peek();
+            if (file != null) {
+                meta = new FileMetadata(
+                        file.fileId(),
+                        Timestamp.newBuilder().seconds(file.expirationTime()).build(),
+                        file.keys(),
+                        file.contents(),
+                        file.memo(),
```

Comment:
As noted, I think we should be getting this metadata from the normal `KVStore`.

There's also an important detail about the `memo` field we return for file `0.0.150` in this query.

It is not an actual memo, but instead the SHA-384 hash of the current contents (see [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/primitives/StateView.java#L386) in `mono-service`.)

Reply:
fixed thanks